connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -361,7 +362,7 @@ private ConnectorTaskId parseConnectorTaskId(String key) {
         }
     }
 
-    private void readConnectorStatus(String key, byte[] value) {
+    private synchronized void readConnectorStatus(String key, byte[] value) {
```

Comment:
`readConnectorStatus` and `readTaskStatus` have calls to `parseConnectorStatus` and `parseTaskStatus` had internal calls to `converter`. Will revert since the calls to the cache tables are synchronized already.